### DISCUSSION TITLE
Playnite launcher configuration

### DISCUSSION THREAD

**Discussion by skarabaraks** [LizardByte/community#113] [Sunshine]
Does anybody know how to configure the Sunshine "Applications" tab so that [Playnite ](https://playnite.link/)launches directly from Moonlight?

My current configuration (below) gives me an error. What am I doing wrong?

Thanks

![Playnite_error2](https://user-images.githubusercontent.com/2365359/208628621-47ce7844-5e9a-4f62-9560-9ff1d092ee74.png)

![Screenshot_20221220-090555](https://user-images.githubusercontent.com/2365359/208627499-fd7e79e4-7e5b-4f8c-aead-659386c78096.png)

![Screenshot_20221220-090405_2](https://user-images.githubusercontent.com/2365359/208627225-35380f8b-5aa5-4cbb-9714-a69e85554277.png)

**Comment by KuleRucket** [comment]
Some suggestions:
- check the log output
- If you are running sunshine as a service, stop the service and run from a window. The service launches applications as SYSTEM user and not the logged in user (known issue) so Playnite might not be looking in the "scarab\AppData\..." folder for it's configuration.  You can also try to get the service to run as the local user.
- also, you are running it as detached so each time you connect it will attempt to run a new instance of Playnite. This may or may not be a problem but worth trying as a "Command". 

See also:
https://lizardbyte--574.org.readthedocs.build/projects/sunshine/en/574/about/usage.html#application-list

**Reply by skarabaraks** [reply]
Thank you for your quick response and the helpful suggestions. 

The "command" option did it

It works perfectly now. Thank you!!!

**Reply by Silas888** [reply]
Let me know how that option works for you. I too, was using the command option. The issue with me though is that Moonlight will launch PlayNite and then immediately lose connection. Moonlight doesn't think anything is running. However, if I look on my PC PlayNite is running. The only way to get Moonlight to connect again is to kill PlayNite from the PC.

**Reply by skarabaraks** [reply]
I've been using the "desktop" shortcut in moonlight with no issues for a while.  

Moonlight supports "mouse mode" when you launch the "desktop shortcut".

I would just press and hold the Start button to enable mouse emulation. 

When mouse emulation is on, I was using the controller analog stick to move the cursor. (A button left clicks and the B button right clicks). 

Then press and hold start button to disable mouse emulation and play the game.

Probably not the most efficient way, but worked well before I figured out how to add the Application shortcuts in sunshine.

Good luck :)

**Reply by skarabaraks** [reply]
> Let me know how that option works for you. I too, was using the command option. The issue with me though is that Moonlight will launch PlayNite and then immediately lose connection. Moonlight doesn't think anything is running. However, if I look on my PC PlayNite is running. The only way to get Moonlight to connect again is to kill PlayNite from the PC.

Also worth checking that when launching Playnite (Playnite.DesktopApp.exe) it doesn't automatically switch to the fullscreen mode of Playnite (Playnite.FullscreenApp.exe). There are 2 of them

**Reply by Silas888** [reply]
> Also worth checking that when launching Playnite (Playnite.DesktopApp.exe) it doesn't automatically switch to the fullscreen mode of Playnite (Playnite.FullscreenApp.exe). There are 2 of them

Well, that's the thing, I have been running it in full screen app. The idea was to have Moonlight launch Playnite. There is, however, some latency delay when using Playnite via Moonlight with the Sunshine host vs. the GFE Host.

**Reply by skarabaraks** [reply]
Perhaps reduce your moonlight resolution + bitrate to improve latency?

I found that sunshine performs better than GFE and certainly better than the AMD version (I'm currently on Radeon RX6800XT). 

My latency hovers around 5ms at 1080p@60fps (SHIELD TV Hardwired ethernet). Hardly if not at all noticeable

**Reply by Silas888** [reply]
> Perhaps reduce your moonlight resolution + bitrate to improve latency?

I did that, but it didn't change anything. But the point is is that it works perfectly fine in GFE but not in Sunshine. That would tell me that there are inherent differences between the two and Sunshine still needs to catch up

**Reply by lmgarret** [reply]
I think I had the same issue with playnite, where opening some of the menus would always be one frame behind. This is similar to what was described here https://github.com/JosefNemec/Playnite/issues/3163
From what I understand, this should be fixed once this (already merged) fix https://github.com/LizardByte/Sunshine/pull/507 is released

**Reply by Silas888** [reply]
That's wonderful. I can see there's going to be a lot of good stuff coming out with a new release of Sunshine. I can't wait.

**Comment by jimmt41** [comment]
when this is merged (https://github.com/LizardByte/Sunshine/discussions/521) this will allow us to properly have a Playnite shortcut and launch with the user configuration.

**Reply by ReenigneArcher** [reply]
#600 you mean

**Comment by demarey-baker** [comment]
Had this happen to me and I had to install nircmd on my PC and it worked.

**Reply by aagustyana** [reply]
hi could you explain how to do that ?

**Comment by goblinzone** [comment]
I got it working on my machine!!

I set Command as:
`"C:\Users\<user>\AppData\Local\Playnite\Playnite.FullscreenApp.exe" --hidesplashscreen`
(a full list of other possible arguments can be found [here](https://playnite.link/docs/manual/cmdlineArguments.html))

e: I've been trying some of the other startup commands out and it seems using `--shutdown` alongside this will counteract the fix.

**Comment by foxy82** [comment]
I had setup a Playnite launcher config with a Detached Command of: "C:\Users\neilr\AppData\Local\Playnite\Playnite.FullscreenApp.exe"

Every 10 minutes or so the game I was playing would minimize and I'd be back at the desktop.

Changing to use "Command" with ""C:\Users\<user>\AppData\Local\Playnite\Playnite.FullscreenApp.exe" --hidesplashscreen" seems to have helped for me.

**Comment by AbrahamHuitron** [comment]
Anyone have some advice on how to get play night to run for Moonlight when the desktop app is already running or minimizing the task tray? I have the full screen app as the shortcut for Sunshine, but if I attempt to run the application while the desktop app is minimized or already running on my PC, I get a connection error. I'm sure there are some very easy command or detached scripts to attach to the application. Anyone know what they are?

Note: I'm trying to get this to run from the command that way when I start the full screen app, the streaming starts, and when it ends, the streaming stops. If I run Moonlight from the detached command, the streaming remains after I closed the full screen app, and I don't want that.

**Comment by PrecedentPowers** [comment]
This command using ''--hidesplashcreen" works, IF Playnite is not already open in desktop mode.  If Playnite is open in desktop mode, then the stream will fail citing a DRM error.  Connecting through Desktop shows that PlayniteFullscreen is open.  I haven't figured out a way to fix this yet.

**Comment by altheir** [comment]
I created a file `playnite.bat`  with the following command:

`TASKLIST | FINDSTR Playnite.FullscreenApp && taskkill /im Playnite.FullscreenApp.exe && timeout /t 3 || exit 0` 

And set it as a `do command`  as part of the command preparations before the actual command of `Playnite.FullscreenApp.exe --hidesplashscreen`. This allowed me to launch it even if I had a fullscreen instance already running on the machine.

**Reply by jimmt41** [reply]
I have something similar. I use bat file to run a PowerShell script. 
Bat file
```bat
@ECHO OFF
REM @ECHO Starting Playnight Full Screen launcher
SET ThisScriptsDirectory=%~dp0
SET PowerShellScriptPath=%ThisScriptsDirectory%sunshineplaynite.ps1
PowerShell -NoProfile -ExecutionPolicy Bypass -Command "& '%PowerShellScriptPath%'";
@exit
```

PShell:
```pwsh
$playnite = %{
    $checkFullScreen = get-process "Playnite.FullscreenApp" -ErrorAction SilentlyContinue; 
    $checkDesktop = get-process "Playnite.DesktopApp" -ErrorAction SilentlyContinue;
    if ($checkFullScreen) {
        Write-Output "already running";
        }
    else {
        if ($checkDesktop) {
            Stop-Process -Force $checkDesktop.id;
            sleep -Seconds 5;
            }
    }
}
if(!$playnite) {
   Set-Location "G:\Playnite\"
   cmd /c "G:\Playnite\Playnite.FullscreenApp.exe"
   } 
else {Write-Output "running"};
```

**Comment by Eyji** [comment]
> This command using ''--hidesplashcreen" works, IF Playnite is not already open in desktop mode. If Playnite is open in desktop mode, then the stream will fail citing a DRM error. Connecting through Desktop shows that PlayniteFullscreen is open. I haven't figured out a way to fix this yet.

>I created a file playnite.bat with the following command:
TASKLIST | FINDSTR Playnite.FullscreenApp && taskkill /im Playnite.FullscreenApp.exe && timeout /t 3 || exit 0
And set it as a do command as part of the command preparations before the actual command of Playnite.FullscreenApp.exe --hidesplashscreen. This allowed me to launch it even if I had a fullscreen instance already running on the machine.

Actually, using the following Command is enough to switch an active Desktop instance to a Fullscreen one, if any.
Otherwise, it will just start Playnite normally

`"C:\Users\<username>\AppData\Local\Playnite\Playnite.FullscreenApp.exe" --hidesplashscreen --startfullscreen`

Source: https://api.playnite.link/docs/master/manual/cmdlineArguments.html

**Reply by buthig666** [reply]
Command:  "C:\Users\<username>\AppData\Local\Playnite\Playnite.FullscreenApp.exe" --hidesplashscreen --startfullscreen

This command works for me but only as Playnite.DesktopApp.exe,  unsure why.  I can configure Sun/Play several ways to open in fullscreen but no matter what I do... if Moonlight opens Playnite up in Fullscreen mode it cannot close it.  No matter what.  I always have to manually close Playnite from Fullscreen.  Moonlight does however close Playnite, if it opens in desktop mode.

I can't figure it out.  There is a --shutdown command listed here (https://api.playnite.link/docs/master/manual/advanced/cmdlineArguments.html)
 but I am not sure how to apply it.  Any ideas?

**Reply by beckman924** [reply]
Did you find a solution? I´m facing the same problem I can´t exit Playnite when I want to close Moonlight streaming

**Reply by buthig666** [reply]
Nope, I just got comfortable with adding apps to sunshine manually.

[a comment by an automated account is not shown]
